test/unit/scripts/upgrade/TimeConstraints.t.sol

```diff
@@ -34,97 +34,182 @@ contract TimeConstraintsTest is Test {
+
+        vm.warp(BASE_TIMESTAMP + currentTime.toSeconds());
 
         vm.expectRevert(
-            abi.encodeWithSelector(
-                TimeConstraints.DayTimeOutOfRange.selector,
-                Durations.from(10 hours),
-                Durations.from(12 hours),
-                Durations.from(18 hours)
-            )
+            abi.encodeWithSelector(TimeConstraints.DayTimeOutOfRange.selector, currentTime, startTime, endTime)
         );
+        timeConstraints.checkTimeWithinDayTime(startTime, endTime);
+    }
 
-        timeConstraints.checkExecuteWithinDayTime(Durations.from(12 hours), Durations.from(18 hours));
+    function testFuzz_checkTimeWithinDayTime_OutOfRange_Overnight(
+        Duration startTime,
+        Duration endTime,
+        Duration currentTime
+    ) public {
+        vm.assume(startTime > endTime);
+        vm.assume(startTime.toSeconds() < 24 hours);
+        vm.assume(endTime.toSeconds() < 24 hours);
+        vm.assume(currentTime.toSeconds() < 24 hours);
+        vm.assume(currentTime > endTime && currentTime < startTime);
 
-        vm.warp(baseTimestamp + 20 hours);
+        vm.warp(BASE_TIMESTAMP + currentTime.toSeconds());
 
         vm.expectRevert(
-            abi.encodeWithSelector(
-                TimeConstraints.DayTimeOutOfRange.selector,
-                Durations.from(20 hours),
-                Durations.from(12 hours),
-                Durations.from(18 hours)
-            )
+            abi.encodeWithSelector(TimeConstraints.DayTimeOutOfRange.selector, currentTime, startTime, endTime)
         );
-
-        timeConstraints.checkExecuteWithinDayTime(Durations.from(12 hours), Durations.from(18 hours));
+        this.external__checkTimeWithinDayTime(startTime, endTime);
     }
 
-    function test_checkExecuteWithinDayTime_TimeOverflow() external {
+    function test_checkTimeWithinDayTime_TimeOverflow() external {
         vm.expectRevert(abi.encodeWithSelector(TimeConstraints.DayTimeOverflow.selector));
-        timeConstraints.checkExecuteWithinDayTime(Durations.from(0), Durations.from(24 hours));
+        this.external__checkTimeWithinDayTime(Durations.from(0), Durations.from(24 hours));
 
         vm.expectRevert(abi.encodeWithSelector(TimeConstraints.DayTimeOverflow.selector));
-        timeConstraints.checkExecuteWithinDayTime(Durations.from(30 hours), Durations.from(12 hours));
+        this.external__checkTimeWithinDayTime(Durations.from(30 hours), Durations.from(12 hours));
     }
 
-    function test_checkExecuteWithinDayTime_InvalidRange() public {
-        vm.expectRevert(
-            abi.encodeWithSelector(
-                TimeConstraints.InvalidDayTimeRange.selector, Durations.from(18 hours), Durations.from(12 hours)
-            )
-        );
+    // checkTimeWithinDayTimeAndEmit
+
+    function testFuzz_checkTimeWithinDayTimeAndEmit_HappyPath_Regular(
+        Duration from,
+        Duration to,
+        Duration timeToWarp
+    ) external {
```

Comment:
More reliably, call `testFuzz_checkTimeWithinDayTime_HappyPath(from, to, timeToWarp)` together with checking that an event is emitted.

Reply:
Same for the other tests which check emitted events

Reply:
It would also be nice to have tests for the edge cases, for example, `from == to`, `from = 24 hours - 1, to = 0`, `to = 0, from = 24 hours - 1`, etc.